### DISCUSSION TITLE
node 2D panes in 3d scenes

### DISCUSSION THREAD

**Discussion by FoxbitDreamtail** [godotengine/godot-proposals#7749] [3D]
There are several possibilities for this like kind of a Mario Odyssey 2D section like area might come to mind or adding 3D effects to a 2d game or adding 2D games into a 3D arcade in a VR scene, I'm just saying it would be ideal for me just to add 2d panes then to try making a whole 3D thing just to flatten it or add a custom texture to a block which would increase the number of things in a scene and make it far more complex than it needs to be.

I just thought about it and you could also have a property to where you can enable 3D scaling so that it scales properly depending on how large you put the viewport in a 3D space to make more 2D sections and 3D games feel smoother then to have to do some guesswork. We could also theoretically also do 3D warping to create Catherine ray tubes or CRTs in games.

**Comment by Zireael07** [comment]
You can already put 2d nodes on meshes (see 2d in 3d demo)

**Reply by FoxbitDreamtail** [reply]
I just figured, wouldn't it be easier to be able to just place a 2d pane in a 3D world and manipulate it in that way? Not just doesn't match but as a kind of placeable thing in 3D space?

**Reply by Zireael07** [reply]
2d and 3d are separate things. You can't put a 2d thing in 3d as 3d has no concept of 2d. You need a mesh - as simple as a quad - to render anything

**Reply by FoxbitDreamtail** [reply]
I'm sorry for my ignorance I'm kind of new to Godot, and Game Dev as a whole, so as for meshes, where would I find the demo you mentioned?

And somewhat more critically, can't you just have it default to creating a mesh with the 2D scene within it if you add it as a child of a 3D node? Wouldn't that be easier than just having it done manually?

**Reply by Zireael07** [reply]
Demo: https://github.com/godotengine/godot-demo-projects/tree/master/viewport/2d_in_3d (or on Asset Store)

Re adding a mesh automatically  - that's actually a nice feature request.

**Reply by FoxbitDreamtail** [reply]
I decided to use the steam version of Godot on my PC, for some reason the asset store is completely broken on the steam version, and I'm just talking from memory again, I'm not near my PC. I'll check it out once I'm near my PC again.
